third_party/terraform/tests/resource_dns_resource_dns_record_set_test.go

```diff
@@ -0,0 +1,312 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestIpv6AddressDiffSuppressForDNS(t *testing.T) {
+	cases := map[string]struct {
+		Old, New       string
+		ShouldSuppress bool
+	}{
+		"compact form should suppress diff": {
+			Old:            "2a03:b0c0:1:e0::29b:8001",
+			New:            "2a03:b0c0:0001:00e0:0000:0000:029b:8001",
+			ShouldSuppress: true,
+		},
+		"different address should not suppress diff": {
+			Old:            "2a03:b0c0:1:e00::29b:8001",
+			New:            "2a03:b0c0:0001:00e0:0000:0000:029b:8001",
+			ShouldSuppress: false,
+		},
+	}
+
+	for tn, tc := range cases {
+		shouldSuppress := AddressDiffSuppress("", tc.Old, tc.New, nil)
+		if shouldSuppress != tc.ShouldSuppress {
+			t.Errorf("%s: expected %t", tn, tc.ShouldSuppress)
+		}
+	}
+}
+
+func TestAccDNSResourceRecordSet_basic(t *testing.T) {
+	t.Parallel()
+
+	zoneName := fmt.Sprintf("dnszone-test-%s", randString(t, 10))
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDnsResourceRecordSetDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDnsResourceRecordSet_basic(zoneName, "127.0.0.10", 300),
+			},
+			{
+				ResourceName:      "google_dns_resource_dns_record_set.foobar",
+				ImportStateId:     fmt.Sprintf("%s/%s.hashicorptest.com./A", zoneName, zoneName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Check both import formats
+			{
+				ResourceName:      "google_dns_resource_dns_record_set.foobar",
+				ImportStateId:     fmt.Sprintf("%s/%s/%s.hashicorptest.com./A", getTestProjectFromEnv(), zoneName, zoneName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccDNSResourceRecordSet_modify(t *testing.T) {
```

Comment:
For consistency with other tests, can we rename this to `TestAccDNSResourceRecordSet_update`?